### PR TITLE
Refactor FeatFinderCLI and add unittests

### PR DESCRIPTION
**Pull request by gedeck**
FeatFinderCLI is one of the scripts included in RDkit. This PR refactors the script by switching from getopt to argparse and separating the parser and execution of the code. This allows writing a Unittest for the script.

### CODE REVIEW

File: rdkit/Chem/FeatFinderCLI.py [hunk 1]
```diff
@@ -37,79 +26,85 @@ def GetAtomFeatInfo(factory, mol):
...
-  except Exception:
-    logger.error("Could not parse Fdef file %s." % fdefFilename, exc_info=True)
-    sys.exit(-1)
 
-  smilesFilename = extras[1]
-  if not os.path.exists(smilesFilename):
-    logger.error("Smiles file %s does not exist." % smilesFilename)
-    sys.exit(-1)
+_splashMessage = """
+-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
```
Comment (greglandrum) [inline rdkit/Chem/FeatFinderCLI.py]: This is a horrible historical artifact.
Please change it to something neutral and correct like:

```
-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
FeatFinderCLI
  Part of the RDKit (http://www.rdkit.org)
-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
```

File: rdkit/Chem/UnitTestFeatFinderCLI.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  def test_FeatFinderCLI(self):
+    smilesFile = os.path.join(RDConfig.RDDataDir, 'NCI', 'first_5K.smi')
+    featureFile = os.path.join(RDConfig.RDCodeDir, 'Chem', 'Pharm2D', 'test_data',
+                               'BaseFeatures.fdef')
+    parser = FeatFinderCLI.initParser()
+    cmd = '-n 10 {0} {1}'.format(featureFile, smilesFile)
+    with outputRedirect() as (out, err):
+      args = parser.parse_args(cmd.split())
+      FeatFinderCLI.processArgs(args, parser)
+    self.assertIn('Mol-1', out.getvalue())
```
Comment (greglandrum) [inline rdkit/Chem/UnitTestFeatFinderCLI.py]: Not sure if this is reasonable or not, but the repeated calls to out.getvalue() seem strange to me. Wouldn't it be better to read that once into a variable and then test against that?

File: rdkit/Chem/UnitTestFeatFinderCLI.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
+from contextlib import contextmanager
```
Comment (greglandrum) [inline rdkit/Chem/UnitTestFeatFinderCLI.py]: Thanks for adding the new test!

File: rdkit/Chem/UnitTestFeatFinderCLI.py (hunk 2 above)
Comment (gedeck) [inline rdkit/Chem/UnitTestFeatFinderCLI.py]: Only reason is lazyness. I will assign to a string and test that.

**Comment by greglandrum** [comment]
Thanks Peter.

**Pull request merged**